pkg/storageos/configmap.go

```diff
@@ -121,19 +126,89 @@ const (
 	tlsEtcdCertsVolume = "etcd-certs"
 )
 
-// createService creates a ConfigMap to store the node container configuration.
-func (s *Deployment) createConfigMap() error {
+// ensureConfigMap creates or updates a ConfigMap to store the node container
+// configuration.
+func (s *Deployment) ensureConfigMap() error {
 	config := configFromSpec(s.stos.Spec, CSIV1Supported(s.k8sVersion))
 
 	labels := make(map[string]string)
 
+	existing, err := s.k8sResourceManager.ConfigMap(configmapName, s.stos.Spec.GetResourceNS(), labels, config).Get()
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	// If there is an existing CM, check for changes and apply any that are set
+	// dynamically via the API.
+	if existing != nil {
+		// Don't update if unchanged.
+		if reflect.DeepEqual(config, existing) {
```

Comment:
How can a `map[string]string` and `*v1.ConfigMap` ever be deeply equal? Would you not need to turn the configMap into a map in order to compare them?

Reply:
doh!  Very well spotted.  It needs to compare with `existing.Data`.  I'm adding a unit test for the function...

Reply:
So that test became a bit more involved than I expected...